modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminDisplayContext.java

```diff
@@ -668,7 +669,15 @@ private SearchContainer<RepositoryEntry> _getDLSearchContainer()
 
 		long repositoryId = getRepositoryId();
 
-		long categoryId = ParamUtil.getLong(_httpServletRequest, "categoryId");
+		long categoryId = GetterUtil.getLong(
+			HttpComponentsUtil.getParameter(
+				PortalUtil.getCurrentURL(_httpServletRequest),
+				"p_r_p_categoryId", true));
+
+		if (categoryId == 0) {
+			categoryId = ParamUtil.getLong(_httpServletRequest, "categoryId");
+		}
+
```

Comment:
This looks funny. It should not be necessary to read public render parameters in this way. Why are you doing this?

Reply:
@adolfopa Because the category URL when pagination, it is always escaped URL.

Reply:
Hi @adolfopa ,
Are any updates for this ticket?
Thanks,
Loc